### PR TITLE
Date strptime

### PR DESCRIPTION
**Pull request by kwerle**
Added Dockerfile & Makefile to make testing easy & consistent.
Tweaked Gemfile so that tests pass (updated activesupport reference).
Added require pry to facilitate debugging.

The actual fix is copied from https://github.com/travisjeffery/timecop/pull/217

### CODE REVIEW

File: test/timecop_test.rb [hunk 1]
```diff
@@ -539,6 +539,26 @@ def test_date_strptime_without_specifying_format
...
+      assert_equal Date.strptime('Thursday', '%A'), Date.new(1984, 3, 1)
+      assert_equal Date.strptime('Monday', '%A'), Date.new(1984, 2, 27)
+    end
+  end
+
+  def test_date_strptime_with_invalid_date
+    begin
+      Date.strptime('', '%Y-%m-%d')
+    rescue ArgumentError => e
+      assert_equal 'invalid date', e.message
```
Comment (nerdrew) [inline test/timecop_test.rb]: Comment from my PR: https://github.com/travisjeffery/timecop/pull/217/files#r228704844

Sorry I dropped the ball on my fix! Thanks for picking it up!

**Comment by travisjeffery** [comment]
This looks pretty good, will give it a good look shortly.

**Comment by kwerle** [comment]
Happy to help in any way I can!

**Comment by dmitry** [comment]
Awesome!

**Pull request merged**